src/clj/rems/db/user_settings.clj

```diff
@@ -25,20 +26,14 @@
       validate-user-settings
       json/generate-string))
 
-(s/defschema PartialUserSettings
-  (map-keys s/optional-key UserSettings))
-
-(def ^:private coerce-partial-user-settings
-  (coerce/coercer! PartialUserSettings json/coercion-matcher))
+(def ^:private coerce-user-settings
+  (coerce/coercer! UserSettings json/coercion-matcher))
 
 (defn- json->settings [json]
-  ;; Allows missing keys, so we don't need to write migrations
-  ;; if we add new keys to user settings. Migrations are needed if
-  ;; we remove keys or make their validation stricter.
   (when json
     (-> json
         json/parse-string
-        coerce-partial-user-settings)))
+        coerce-user-settings)))
```

Comment:
So now if some user didn't have a `:language` or `:notification-email` this will fail. It seems that we don't populate user settings when creating a user, so this will probably break production.
```
rems=> (rems.db.users/add-user! {:userid "settings-test"})
(1)
rems=> (rems.db.core/get-user-settings {:user "settings-test"})
nil
```

One solution would be to use a PartialUserSettings-like schema for the db storage, and produce something that matches GetUserSettings (i.e. with guaranteed language and notification-email) from get-user-settings.

Reply:
Oh, and in case we don't have a test that fetched the user settings of a freshly created user, we probably should have one, since it would've caught this.

Reply:
Ah, I misread the code a bit. The `when json` handles empty user settings, but what about a case where somebody has set _just the language_ via the API. Then this code will fail due to a missing `:notification-email` key.

Reply:
Hmm, even that seems to work since we merge `(get-user-settings user)` with the new settings. This is subtle.

Reply:
We do have a test case for a new user in `test-user-settings`.

Reply:
Yeah I'm pretty sure this would be safe to merge. Just needed to think through the logic.